### PR TITLE
[bitnami/kong] MAJOR - Kong subchart bump

### PR DESCRIPTION
**Pull request by joancafom**
**Description of the change**

This PR bumps Kong major version since its dependencies (both PostgreSQL and Cassandra) bumped their major versions including breaking changes. More info at #3021 (PostgreSQL) and (Cassandra) #3805.

It also introduces some common standardizations found across charts, such as affinityPressets and common labels & annotations.

**Benefits**

Up-to-date subcomponents, standardization 

**Possible drawbacks**

Breaks backwards compatibility

**Applicable issues**
  NA

**Additional information**
This PR uses some of the validations introduced in #4110 , should merge those changes and update `requirements.lock` before merging this one

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/kong/README.md
Comment (juan131) [inline bitnami/kong/README.md:425]: > Depending on your backend

What do you mean? Please elaborate it properly.

---

Does Kong provides its own way to bump/restore the data?

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the PR!! Please take a look to my comments

File: bitnami/kong/requirements.lock
Comment (joancafom) [reply bitnami/kong/requirements.lock:12]: This PR uses some of the validations introduced in #4110 so I was waiting until merging those ones in order to update `requirements.lock`. Thanks for pointing it out though 😄

File: bitnami/kong/README.md
Comment (joancafom) [reply bitnami/kong/README.md:425]: Kong could be integrated with either PostgreSQL or Cassandra as per user preference. The property `database` in values lets you choose between this two. I will update the README to make this more clear 👍.

> Does Kong provides its own way to bump/restore the data?

To the best of my knowledge, it does not provide with its own feature to restore backup/restore the data. As our Kong setup relies on DBs to store configuration, restoring them would be sufficient.

File: bitnami/kong/requirements.lock
Comment (juan131) [reply bitnami/kong/requirements.lock:12]: In this case, I'd block this PR until then

**Comment by joancafom** [comment]
Changes implemented @juan131

File: bitnami/kong/README.md
Comment (juan131) [inline bitnami/kong/README.md:89]: Add this section to the README.md:
```
### Setting Pod's affinity

This chart allows you to set your custom affinity using the `affinity` paremeter. Find more infomation about Pod's affinity in the [kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity).

As an alternative, you can use of the preset configurations for pod affinity, pod anti-affinity, and node affinity available at the [bitnami/common](https://github.com/bitnami/charts/tree/master/bitnami/common#affinities) chart. To do so, set the `podAffinityPreset`, `podAntiAffinityPreset`, or `nodeAffinityPreset` parameters.
```

File: bitnami/kong/templates/deployment.yaml
Comment (juan131) [inline bitnami/kong/templates/deployment.yaml:56]:
```suggestion
        podAffinity: {{- include "common.affinities.pods" (dict "type" .Values.podAffinityPreset "component" "server" "context" $) | nindent 10 }}
        podAntiAffinity: {{- include "common.affinities.pods" (dict "type" .Values.podAntiAffinityPreset "component" "server" "context" $) | nindent 10 }}
```

**Review by juan131 — APPROVED** [review]
Approving your PR! Please implement these two suggestions before merging it

**Pull request merged**
